client/src/main/java/io/pravega/client/batch/impl/SegmentIteratorImpl.java

```diff
@@ -78,13 +90,27 @@ public T next() {
                                        }
                                        return buffer;
                                    } catch (NoSuchSegmentException | SegmentTruncatedException e) {
+                                       handleSegmentTruncated(segment);
                                        throw new TruncatedDataException("Segment " + segment + " has been truncated.");
                                    }
                                });
 
         return deserializer.deserialize(read);
     }
 
+    private void handleSegmentTruncated(Segment segmentId)  {
+        log.info("{} encountered truncation for segment while read{} ", this, segmentId);
+        @Cleanup
+        SegmentMetadataClient metadataClient = segmentMetadataClientFactory.createSegmentMetadataClient(segmentId,
+                DelegationTokenProviderFactory.create(controller, segmentId, AccessOperation.READ));
+        try {
+            long startingOffset = Futures.getThrowingExceptionWithTimeout(metadataClient.getSegmentInfo(), timeoutInMilli).getStartingOffset();
+            input.setOffset(startingOffset);
+        } catch (TimeoutException te) {
+            log.warn("A timeout has occurred while attempting to retrieve segment information from the server");
```

Comment:
Can add info about segment in warning log for better clarity

Reply:
done